src/App.tsx

```diff
@@ -14,11 +14,79 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+// Types are required by TypeScript Engine for the App Class Component
+type Props = {};
+type State = {
+  selectedGood: string,
+  startIndex: number,
+};
+
+class App extends React.Component<Props, State> {
+  state = {
+    selectedGood: 'Jam',
+    startIndex: -1,
+  };
+
+  /* This method clears a product name in the <h1> title
+  if it`s selected, deselects product and renders a Select <button> */
+  productTitle = () => {
+    this.setState({
+      selectedGood: '',
+      startIndex: -1,
+    });
+  };
+
+  render() {
+    const { startIndex, selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          Selected good: -
+          {' '}
+          {this.state.selectedGood
+            ? `${selectedGood} is selected`
+            : 'No goods selected'}
+          <button
+            type="button"
+            onClick={this.productTitle}
+          >
+            X
+          </button>
+        </h1>
+
+        <ul>
+          {goodsFromServer.map((good, index) => (
+            <>
+              <li
+                key={good}
+                className={`item ${startIndex !== index ? '' : 'active'}`}
```

Comment:
```suggestion
                className={`item ${startIndex !== index ? '' : 'active'}`}
```
Для динамічного використання класів в реакті, доречніше використовувати бібліотеку classnames
https://www.npmjs.com/package/classnames